exp_to_json.py

```diff
@@ -54,32 +132,69 @@ def exp_to_json(file_path):
                 so_obj_type = re.search(r'so_obj_type=([^\s]*)', line).group(1)
-                exp_json['so_object_cond'].append({
-                    'so_module': so_module,
-                    'so_soc_order': so_soc_order,
-                    'so_obj_type': so_obj_type,
-                    'params': {}
+
+                if so_task_name not in exp_json['so_object_cond']:
+                    exp_json['so_object_cond'][so_task_name] = {
+                        'so_module': so_module,
+                        'so_obj_type': so_obj_type,
+                        'conditions': []
+                    }
+
+                exp_json['so_object_cond'][so_task_name]['conditions'].append({
+                    'so_soc_order': so_soc_order
                 })
+
                 continue
 
-            # END reset
+            # reset all switches if END
             if line.startswith('END'):
-                is_job_table = is_job_prompts = is_object_cond = False
+                _reset_switches()
+
+            # **********************************************
+            # add matched results to correspoinding sections
+            # **********************************************
 
             match = re.search(r'^(.*?)=(.*)', line)
+
+            # job_table
             if match and is_job_table:
                 if match.group(1) == 'roles':
                     exp_json['so_job_table']['params'][match.group(1)] += match.group(2).split()
                 else:
                     exp_json['so_job_table']['params'][match.group(1)] = match.group(2)
+
+            # documentation
+            elif match and is_documentation:
+                exp_json['so_documentation']['so_doc_text'] += match.group(2)
+
+            # module_sched
+            elif match and is_module_sched:
+                exp_json['aw_module_sched'][module_sched_idx]['params'][match.group(1)] = match.group(2)
+
+            # chain_detail
+            elif match and is_chain_detail:
+                chain_params = exp_json['so_chain_detail'][chain_idx]['params']
+                if match.group(1) == 'so_predecessors':
+                    chain_params[match.group(1)] = re.findall(r'&/(.*?) =', match.group(2))
+                else:
+                    chain_params[match.group(1)] = match.group(2)
+
+            # job_prompts
             elif match and is_job_prompts:
                 exp_json['so_job_prompts'][int(so_prompt)-1]['params'][match.group(1)] = match.group(2)
+
+            # object_cond
             elif match and is_object_cond:
-                exp_json['so_object_cond'][int(so_soc_order)-1]['params'][match.group(1)] = match.group(2)
+                exp_json['so_object_cond'][so_task_name]['conditions'][int(so_soc_order)-1][match.group(1)] = match.group(2)
 
             # TODO: need to handle line starts with 'DELETE'?
 
     exp_json['checks'] = check_dict
+    # import json
+    # print(json.dumps(exp_json, indent=4))
```

Comment:
I'm guessing this was just here for testing. Maybe remove it or add a command line argument to do this dump if debug mode is enabled, for example?